File: selector/src/main/java/com/luck/picture/lib/language/LocaleTransform.java
package com.luck.picture.lib.language;

import java.util.Locale;

/**
 * @author：luck
 * @date：2019-11-25 21:58
 * @describe：语言转换
 */
public class LocaleTransform {
    public static Locale getLanguage(int language) {
        switch (language) {
            case LanguageConfig.ENGLISH:
                // 英语-美国
                return Locale.ENGLISH;
            case LanguageConfig.TRADITIONAL_CHINESE:
                // 繁体中文
                return Locale.TRADITIONAL_CHINESE;
            case LanguageConfig.KOREA:
                // 韩语
                return Locale.KOREA;
            case LanguageConfig.GERMANY:
                // 德语
                return Locale.GERMANY;
            case LanguageConfig.FRANCE:
                // 法语
                return Locale.FRANCE;
            case LanguageConfig.JAPAN:
                // 日语
                return Locale.JAPAN;
            case LanguageConfig.VIETNAM:
                // 越南语
                return new Locale("vi");
            case LanguageConfig.SPANISH:
                // 西班牙语
                return new Locale("es", "ES");
            case LanguageConfig.PORTUGAL:
                // 葡萄牙语
                return new Locale("pt", "PT");
            case LanguageConfig.AR:
                // 阿拉伯语
                return new Locale("ar", "AE");
            case LanguageConfig.RU:
                // 俄语
                return new Locale("ru", "rRU");
            case LanguageConfig.CS:
                // 捷克
                return new Locale("cs", "rCZ");
            case LanguageConfig.KK:
                // 哈萨克斯坦
                return new Locale("kk", "rKZ");
            default:
                // 简体中文
                return Locale.CHINESE;
        }
    }
}

File: selector/src/main/java/com/luck/picture/lib/language/LocaleTransform.kt
package com.luck.picture.lib.language

import java.util.*

/**
 * @author：luck
 * @date：2019-11-25 21:58
 * @describe：语言转换
 */
object LocaleTransform {
    fun getLanguage(language: Language): Locale {
        return when (language) {
            Language.ENGLISH ->                 // 英语-美国
                Locale.ENGLISH
            Language.TRADITIONAL_CHINESE ->                 // 繁体中文
                Locale.TRADITIONAL_CHINESE
            Language.KOREA ->                 // 韩语
                Locale.KOREA
            Language.GERMANY ->                 // 德语
                Locale.GERMANY
            Language.FRANCE ->                 // 法语
                Locale.FRANCE
            Language.JAPAN ->                 // 日语
                Locale.JAPAN
            Language.VIETNAM ->                 // 越南语
                Locale("vi")
            Language.SPANISH ->                 // 西班牙语
                Locale("es", "ES")
            Language.PORTUGAL ->                 // 葡萄牙语
                Locale("pt", "PT")
            Language.AR ->                 // 阿拉伯语
                Locale("ar", "AE")
            Language.RU ->                 // 俄语
                Locale("ru", "rRU")
            Language.CS ->                 // 捷克
                Locale("cs", "rCZ")
            Language.KK ->                 // 哈萨克斯坦
                Locale("kk", "rKZ")
            else ->                 // 简体中文
                Locale.CHINESE
        }
    }
}